react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+class AppDrawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isShown: false };
+    this.cssClass = 'side-menu';
+    this.handleClick = this.handleClick.bind(this);
+    this.hideMenu = this.hideMenu.bind(this);
+  }
+
+  handleClick(event) {
+    this.setState(function (state) {
+      if (!this.state.isShown) {
+        this.cssClass += ' show';
+        document.body.addEventListener('click', this.hideMenu);
+        document.body.style.backgroundColor = 'rgba(0, 0, 0, 0.4)';
+        return { isShown: true };
+      } else {
+        document.body.style.backgroundColor = 'white';
+        return { isShown: false };
```

Comment:
In React, we 99.99% _never_ query the DOM, or reach outside to other DOM elements. Overall, I believe what your solution is missing is an absolutely positioned "shade" element like a modal background. You can render that element from your component and it can take up the whole page. Check your CSS Modals solution for the CSS

Reply:
Removed the dom queries and added a modal instead.